godot-macros/src/class/derive_godot_class.rs

```diff
@@ -314,20 +312,23 @@ fn parse_struct_attributes(class: &venial::Struct) -> ParseResult<ClassAttribute
             is_editor_plugin = true;
 
             // Requires #[class(tool, base=EditorPlugin)].
-            if !is_tool {
+            if base_ty != ident("EditorPlugin") {
                 return bail!(
                     attr_key,
-                    "#[class(editor_plugin)] requires additional key `tool`"
+                    "#[class(editor_plugin)] requires additional key-value `base=EditorPlugin`"
                 );
             }
-            if base_ty != ident("EditorPlugin") {
+            if !is_tool {
                 return bail!(
                     attr_key,
-                    "#[class(editor_plugin)] requires additional key-value `base=EditorPlugin`"
+                    "#[class(editor_plugin)] requires additional key `tool`"
                 );
             }
```

Comment:
Maybe write a comment about the reasoning behind the order, to avoid that future refactors change it again. You can also link to this PR for details.